src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/operations/blob.py

```diff
@@ -417,6 +417,8 @@ def generate_sas_blob_uri(client, container_name, blob_name, permission=None,
                           protocol=None, cache_control=None, content_disposition=None,
                           content_encoding=None, content_language=None,
                           content_type=None, full_uri=False):
+    if client.account_key is None:
+        raise CLIError("Missing --account-key to generate sas")
```

Comment:
This should probably be in usage string format, but generally, is there a logic bug?  Typically if you specify a storage account but not the key (or other credential) then the CLI will retrieve the key behind the scenes unless unable.

Reply:
Yes, the logic is that CLI will retrieve the key if there is only account name.

But from the call stack,

493:        _validate_not_none('self.account_name', self.account_name)
494:        _validate_not_none('self.account_key', self.account_key)

It failed at 494 line. That mean account_key is None but account_name is not None. So it did happen somehow in such case.

Reply:
In that case, it seems like the problem is in the logic that retrieves the account key. It would make more sense to do this logic there instead of having a special case check right here. It could also be that this command isn't hitting that logic.